packages/create-llama/index.ts

```diff
@@ -234,6 +237,30 @@ async function run(): Promise<void> {
     installDependencies: program.installDependencies,
   });
   conf.set("preferences", preferences);
+
+  if (program.runApp) {
+    try {
+      const cps = await runApp(
+        resolvedProjectPath,
+        program.frontend,
+        program.framework,
+        program.externalPort,
+      );
+
+      // Lock the process until all child processes exit
+      await Promise.all(
+        cps.map(
+          (cp) =>
+            new Promise((resolve, reject) => {
+              cp.on("exit", resolve);
+              cp.on("error", reject);
+            }),
+        ),
+      );
+    } catch (e) {
+      throw new RunApplicationError();
```

Comment:
error handling in this app is not currently by throwing a specific error type.
if you don't want the generic error handler,  how about just logging the error here and calling `process.exit(1)`?

Reply:
you could then also just do here:

```
await runApp(
        resolvedProjectPath,
        program.frontend,
        program.framework,
        program.externalPort,
      );
 ```

and put the process locking into `runApp`

Reply:
I updated the runApp behavior to support better logging and handle child processes as well. Please have a look, thank you!